Print deployment results as a table

jordanstephens opened the pull request:
## Description

Use `term.Table` to print deployed services with their id, name, status, endpoints, and domain names.

## Linked Issues


## Checklist

- [x] I have performed a self-review of my code
- [x] I have added appropriate tests
- [ ] I have updated the Defang CLI docs and/or README to reflect my changes, if necessary

lionello reviewed (changes requested):
I think that table can/should match the one we show for "get services" https://github.com/DefangLabs/defang/blob/d889d1abd2cadf2f7eb1ba9bd09c379596349d5d/src/pkg/cli/getServices.go#L63

* Id vs Etag
* Name vs Service
* State vs Status
* Endpoints vs PublicFqdn+PrivateFqdn

I like the use of "id" vs "etag", but prefer "service" to "name" (less ambiguous). Endpoints can replace the FQDNs.

jordanstephens commented:
> I think that table can/should match the one we show for "get services"
> 
> https://github.com/DefangLabs/defang/blob/d889d1abd2cadf2f7eb1ba9bd09c379596349d5d/src/pkg/cli/getServices.go#L63
> 
>     * Id vs Etag
> 
>     * Name vs Service
> 
>     * State vs Status
> 
>     * Endpoints vs PublicFqdn+PrivateFqdn
> 
> 
> I like the use of "id" vs "etag", but prefer "service" to "name" (less ambiguous). Endpoints can replace the FQDNs.

I updated the service table to use "ID" instead of "etag", and I updated the deployment table to use "status" instead of "state". I am keeping the word "service" and not using the word "name". I am inclined to keep separate public and private fqdn though, because I prefer having two fields with short values, over having one field with a potentially very long comma-delimited list inlined.

lionello commented on src/pkg/cli/getServices.go:
```diff
@@ -21,6 +21,7 @@ func (e ErrNoServices) Error() string {
 
 type printService struct {
 	Service string
+	ID      string
```
```suggestion
	Deployment      string
```

lionello commented on src/pkg/cli/getServices.go:
```diff
@@ -55,10 +56,11 @@ func GetServices(ctx context.Context, projectName string, provider client.Provid
 	}
 
-	return term.Table(printServices, []string{"Service", "Etag", "PublicFqdn", "PrivateFqdn", "Status"})
+	return term.Table(printServices, []string{"Service", "ID", "PublicFqdn", "PrivateFqdn", "Status"})
```
Make sure `Status` is the one we want and not `State`. `State` might be more useful, because it would show things like "out of memory".

lionello commented on src/cmd/cli/command/deploymentinfo.go:
```diff
@@ -20,31 +22,52 @@ func printPlaygroundPortalServiceURLs(serviceInfos []*defangv1.ServiceInfo) {
-			andEndpoints = "and will be available at:"
-		}
+type ServiceTableItem struct {
+	Id         string `json:"Id"`
```
Rename?

lionello commented on src/pkg/cli/getServices_test.go:
```diff
@@ -82,8 +82,8 @@ func TestGetServices(t *testing.T) {
 		}
-		expectedOutput := `Service  Etag    PublicFqdn                 PrivateFqdn  Status
-foo      a1b2c3  test-foo.prod1.defang.dev               UNKNOWN
+		expectedOutput := `Service  Deployment  PublicFqdn                 PrivateFqdn  State
```
This became `State` where I thought it's `Status` we want? Or both.

jordanstephens replied:
You had suggested wanting to use `State` in this comment: https://github.com/DefangLabs/defang/pull/1033#discussion_r2015137274

lionello replied:
Shoot, I mixed the up. State was added for the subscription feature (stop CLI when deployment done), that's why it's an enum. Status has more info.

jordanstephens replied:
https://github.com/DefangLabs/defang/pull/1033/commits/1f76c5e8254303c9413878a2864f35fba5472540

jordanstephens commented on src/cmd/cli/command/commands.go:
```diff
@@ -890,7 +897,7 @@ var debugCmd = &cobra.Command{
 
 		debugConfig := cli.DebugConfig{
-			Etag:           etag,
+			Etag:           deployment,
```
If I start renaming these fields, the diff grows very large. Let me do it in a followup.

jordanstephens commented on src/cmd/cli/command/compose.go:
```diff
@@ -484,7 +492,7 @@ func makeComposeLogsCmd() *cobra.Command {
 
 			tailOptions := cli.TailOptions{
-				Etag:     etag,
+				Etag:     deployment,
```
https://github.com/DefangLabs/defang/pull/1033#discussion_r2019254560
